### PR TITLE
IAM docs: Update Azure AD OAuth2.0 docs after rename to Entra ID

### PR DESCRIPTION
**Pull request by volcanonoodle**
Replaces all mentions of Azure Active Directory with Microsoft Entra ID, which is how the same product is called nowadays.

It keeps mentions of Azure AD OAuth2 and the like when they refer to the Grafana part of the the config, as it's still named Azure AD there.

### CODE REVIEW

**Review by mgyongyosi — APPROVED** [review]
Looks good to me, but please wait for @chri2547's review before merging!

**Review by chri2547 — APPROVED** [review]
Looks great - thank you for the contribution!

**Pull request merged**
